db.go

```diff
@@ -53,13 +53,25 @@ Every time you will run this application, it will remain in the same state as be
 package txdb
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
 	"fmt"
 	"io"
 	"sync"
 )
 
+// New returns a driver.Connector, which can be passed to sql.OpenDB. This can
+// be used in place of Register.
+func New(drv, dsn string, options ...func(*conn) error) driver.Connector {
```

Comment:
Not sure if `New` is the best function name for this.  I'm open to alternative suggestions.